include/rnp/rnp.h

```diff
@@ -1349,6 +1349,81 @@ RNP_API rnp_result_t rnp_key_is_retired(rnp_key_handle_t key, bool *result);
  **/
 RNP_API rnp_result_t rnp_key_is_locked(rnp_key_handle_t key, bool *result);
 
+/**
+ * @brief Get type of protection, used for secret key data.
+ *
+ * @param key key handle, cannot be NULL and should have secret part (see function
+ *            rnp_key_have_secret()).
+ * @param type on success protection type will be stored here. Cannot be NULL.
```

Comment:
When I read this, I thought the buffer should be allocated by the caller. But I see in the source that buffer is allocated using strdup() inside the function. So it should be explicitly mentioned that buffer is allocated by the function and that it should be released by rnp_buffer_destroy().

Also, is there a real reason for duplicating these strings - "None", "Encrypted", etc? They are static string literals, and therefore just pointer to them could be returned instead, and type parameter can be declared as `const char **type`. Is there any need that caller could modify contents of returned buffer later? This optimization could reduce memory usage.

Reply:
@antonsviridenko Thanks for pointing at this, forgot to mention that string should be deallocated. Will fix it.

In all other function calls string constants (like algorithm names) are also duplicated. As I understand, the idea behind this is that caller shouldn't have any direct interpretable pointers to library internal things - all returned values are owned by caller.